lib/Common/Memory/Recycler.cpp

```diff
@@ -5362,6 +5436,49 @@ Recycler::BackgroundMark()
     }
 }
 
+void
+Recycler::BackgroundMarkWeakRefs()
+{
+#if ENABLE_WEAK_REFERENCE_REGIONS
+    auto iterator = this->weakReferenceRegionList.GetIterator();
+    while (iterator.Next())
```

Comment:
I believe so. It wouldn't be safe to iterate if the foreground thread is deleting things, but my understanding of `SList` is that in the worst case we might miss an entry that is being added in the foreground, and that means that we don't mark quite as much as we otherwise could have.